misc/python/materialize/output_consistency/output_consistency.py

```diff
@@ -0,0 +1,171 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+import argparse
+
+import pg8000
+from pg8000 import Connection
+
+from materialize.output_consistency.common.configuration import (
+    ConsistencyTestConfiguration,
+)
+from materialize.output_consistency.common.format_constants import CONTENT_SEPARATOR_1
+from materialize.output_consistency.execution.evaluation_strategy import (
+    ConstantFoldingEvaluation,
+    DataFlowRenderingEvaluation,
+)
+from materialize.output_consistency.execution.sql_executor import create_sql_executor
+from materialize.output_consistency.execution.test_summary import ConsistencyTestSummary
+from materialize.output_consistency.generators.expression_generator import (
+    ExpressionGenerator,
+)
+from materialize.output_consistency.generators.query_generator import QueryGenerator
+from materialize.output_consistency.known_inconsistencies.known_deviation_filter import (
+    KnownOutputInconsistenciesFilter,
+)
+from materialize.output_consistency.runner.test_runner import ConsistencyTestRunner
+from materialize.output_consistency.validation.result_comparator import ResultComparator
+
+
+def run_output_consistency_tests(
+    connection: Connection, args: argparse.Namespace
+) -> ConsistencyTestSummary:
+    """Entry point for output consistency tests"""
+
+    return _run_output_consistency_tests_internal(
+        connection,
+        args.seed,
+        args.dry_run,
+        args.fail_fast,
+        args.execute_setup,
+        args.verbose,
+        args.runtime_in_sec,
+        args.max_iterations,
+        args.avoid_expressions_expecting_db_error,
+    )
+
+
+def parse_output_consistency_input_args(
+    parser: argparse.ArgumentParser,
+) -> argparse.Namespace:
+    parser.add_argument("--seed", default=0, type=int)
+    parser.add_argument(
+        "--dry-run", default=False, type=bool, action=argparse.BooleanOptionalAction
+    )
+    parser.add_argument(
+        "--fail-fast", default=False, type=bool, action=argparse.BooleanOptionalAction
+    )
+    parser.add_argument(
+        "--execute-setup",
```

Comment:
Can we have this also reset the existing data structures?

Reply:
Sure, will do

Reply:
✅